tutorials-v4/time-evolution/012_floquet_formalism.md

```diff
@@ -0,0 +1,276 @@
+
+```python
+psi_t_direct = floquet_wavefunction_t(f_modes0, f_energies, f_coeff, t, H, T)
+assert np.allclose(psi_t.full(), psi_t_direct.full())
+```
+
+### Precomputing and reusing the floquet modes of one period
+
+The Floquet modes have the same periodicity as the Hamiltonian: 
+
+$$ \phi_\alpha(t + T) = \phi_\alpha(t) $$
+
+Hence it is enough to evaluate the modes at times $t \in [0,T]$. From these modes we can extrapolate the system state $\psi(t)$ for any time $t$. 
+
+The function `floquet_modes_table` allows to calculate the floquet modes for multiple times in the first period. 
+
+
+```python
+tlist_period = np.linspace(0, T, 50)
+fmodes_table = floquet_modes_table(f_modes0, f_energies, tlist_period, H, T)
+```
+
+Again, the function `floquet_wavefunction` (introduced above) can be used to build the wavefunction $\psi(t)$. Here, we calculate the expectation value for the number operator in the first period.
+
+```python
+p_ex_period = []
+for i, t in enumerate(tlist_period):
+    psi_t = floquet_wavefunction(fmodes_table[i], f_energies, f_coeff, t)
+    p_ex_period.append(expect(num(2), psi_t))
+plt.plot(tlist_period, p_ex_period)
+plt.ylabel("Occupation prob."), plt.xlabel("Time");
+```
+
+The pre-computed modes for the first period can be used by the function `floquet_modes_t_lookup` to calculate the floquet modes at any time $t > 0$. Note that if a time $t'$ is not exactly $t' = t + nT$ (where $t$ is a time used in the pre-computation) the closest pre-computed floquet mode is used. This might lead to small discontinuoties in the results from the lookup.
+
+```python
+p_ex = []
+tlist = np.linspace(0, 10 * T, 10 * tlist_period.shape[0])
+for n, t in enumerate(tlist):
+    f_modes_t = floquet_modes_t_lookup(fmodes_table, t, T)
+    psi_t = floquet_wavefunction(f_modes_t, f_energies, f_coeff, t)
+    p_ex.append(expect(num(2), psi_t))
+
+# Plot the occupation Probability
+plt.plot(tlist, p_ex, label="Lookup method")
+plt.plot(tlist_period, p_ex_period, label="First period - precomputed")
+plt.legend(loc="upper right")
+plt.xlabel("Time"), plt.ylabel("Occupation prob.");
+```
+
+### Floquet Markov formalism
+
+We can also solve a master equation using the Floquet formalism. A detailed derivation of the Floquet-Markov formalism used here is given in [Grifoni et al., Physics Reports 304, 299 (1998)](https://www.sciencedirect.com/science/article/abs/pii/S0370157398000222) and in the [QuTiP docs](https://qutip.org/docs/latest/guide/dynamics/dynamics-floquet.html). Note that the functionality described here is summarised in the function `fmmesolve` described in the [floquet solver notebook](011_floquet_solver.md).
+
+The interaction with the bath is described by a noise spectrum, which does not include the temperature dependency. Hence, the definition is slightly different to the one in the Bloch-Redfield formalism.  For details see the derivation of the formalism.
```

Comment:
Maybe can include a comment that the temperature can be set using w_th argument?